### PR TITLE
Inline legacyToWriteMapping in Redshift connector

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

**Review by hashhar — APPROVED** [review]

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java [hunk 1]
```diff
@@ -25,43 +26,85 @@
...
+import static io.trino.spi.type.TinyintType.TINYINT;
+import static io.trino.spi.type.VarbinaryType.VARBINARY;
 import static java.lang.Math.max;
 import static java.lang.String.format;
 
 public class RedshiftClient
         extends BaseJdbcClient
 {
+    /**
+     * @deprecated To be removed after {{@link #legacyToWriteMapping} is removed.
```
Comment (findepi) [inline plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:93]: you did remove references to this method, right?

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java [hunk 2]
```diff
@@ -25,43 +26,85 @@
...
+    @Deprecated
+    private static final Map<Type, WriteMapping> WRITE_MAPPINGS = ImmutableMap.<Type, WriteMapping>builder()
+            .put(BOOLEAN, WriteMapping.booleanMapping("boolean", booleanWriteFunction()))
+            .put(BIGINT, WriteMapping.longMapping("bigint", bigintWriteFunction()))
+            .put(INTEGER, WriteMapping.longMapping("integer", integerWriteFunction()))
+            .put(SMALLINT, WriteMapping.longMapping("smallint", smallintWriteFunction()))
+            .put(TINYINT, WriteMapping.longMapping("tinyint", tinyintWriteFunction()))
+            .put(DOUBLE, WriteMapping.doubleMapping("double precision", doubleWriteFunction()))
+            .put(REAL, WriteMapping.longMapping("real", realWriteFunction()))
+            .put(VARBINARY, WriteMapping.sliceMapping("varbinary", varbinaryWriteFunction()))
```
Comment (findepi) [inline plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:104]: this is misleading. it looks like we know what's the proper mapping for redshift, whereas we do not

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:93]: Not yet removed, but I should have changed the comment. Let me change when removing the method.

File: plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java:104]: I thought below comment is sufficient. I will extract a method and move these mapping to there if needed. 
```
L161. // TODO (https://github.com/trinodb/trino/issues/497) Implement proper type mapping and add test
```

**Pull request merged**